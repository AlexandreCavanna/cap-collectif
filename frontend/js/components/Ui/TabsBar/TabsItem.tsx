import * as React from 'react'
import { useIntl } from 'react-intl'
import cn from 'classnames'
import TabsBarDropdown from './TabsBarDropdown'
import * as S from './styles'
import type { Item } from '~/components/Navbar/Navbar.type'

type Props = {
  item: Item
}

const TabsItem = ({ item }: Props) => {
  const intl = useIntl()
  const ariaTitle = `${item.title} - ${intl.formatMessage({
    id: 'active.page',
  })}`

  if (item.hasEnabledFeature) {
    if (item.children && item.children.length > 0) {
      return <TabsBarDropdown item={item} />
    }

    return item.link ? (
      <S.TabsLink
        id={`tabs-navbar-link-${item.id}`}
        href={item.link}
        active={item.active}
        className={cn({
          'tabs-navbar-link-active': item.active,
        })}
        title={item.active ? ariaTitle : null}
      >
        {item.title}
      </S.TabsLink>
    ) : (
      <S.TabsParent id={`tabs-navbar-parent-${item.id}`}>{item.title}</S.TabsParent>
    )
  }

  return null
}

export default TabsItem
