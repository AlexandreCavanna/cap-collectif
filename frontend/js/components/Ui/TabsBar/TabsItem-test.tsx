/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import TabsItem from './TabsItem'

describe('<TabsItem />', () => {
  const props = {
    item: {
      id: 6,
      title: 'Comment Ã§a marche',
      link: '/pages/comment-%C3%A7a-marche',
      hasEnabledFeature: true,
      children: [],
      active: false,
    },
  }
  it('should render', () => {
    const wrapper = shallow(<TabsItem {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
