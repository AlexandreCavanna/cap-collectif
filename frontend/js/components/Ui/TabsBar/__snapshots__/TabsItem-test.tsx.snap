// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsItem /> should render 1`] = `
<styles__TabsLink
  active={false}
  className=""
  href="/pages/comment-%C3%A7a-marche"
  id="tabs-navbar-link-6"
  title={null}
>
  Comment Ã§a marche
</styles__TabsLink>
`;
